src/nupic/math/ArrayAlgo.hpp

```diff
@@ -3786,106 +3786,7 @@ namespace nupic {
       int n2 = (int)(end - start - n1);
 
 
-#if defined(NTA_PLATFORM_darwin86_disabled) || defined(NTA_PLATFORM_linux32_disabled)
```

Comment:
It was a dead code, we did not set the `NTA_PLATFORM_*_disabled` flags anywhere. 
Plus, very similar ASM optimization is applied to 64b right below 
https://github.com/breznak/nupic.core/commit/9810d06dfff8c10626d83d13acd9e5caa4cc6df4?diff=unified#diff-5fc4121d5c281de47186c9f30e425bdbR3798